packages/react-native-reanimated/src/component/ReduceMotionConfig.tsx

```diff
@@ -0,0 +1,33 @@
+'use strict';
+import { useEffect } from 'react';
+import { ReduceMotion } from '../commonTypes';
+import { IsReduceMotion } from '../hook/useReducedMotion';
+
+/**
+ * A component that lets you overwrite default reduce motion behavior globally in your application.
+ *
+ * @param mode - Determines default reduce motion behavior globally in your application. Configured with {@link ReduceMotion} enum.
+ * @see https://docs.swmansion.com/react-native-reanimated/docs/components/ReduceMotionConfig
+ */
+export function ReduceMotionConfig({ mode }: { mode: ReduceMotion }) {
+  useEffect(() => {
+    if (!__DEV__) {
+      return;
+    }
+    console.warn(
+      `[Reanimated] Reduced motion setting is overwritten with mode '${mode}'.`
+    );
+  }, []);
+
+  useEffect(() => {
+    const wasEnabled = IsReduceMotion.jsValue;
+    if (mode === ReduceMotion.Never) {
+      IsReduceMotion.setEnabled(false);
+    }
```

Comment:
What about a situation where someone sets component's mode to `.Always` and system's reduced motion is turned off? The reduced motion will stay OFF. This case is not handled. I'd add an `else if` case for `mode === ReduceMotion.Always`

Reply:
In my assumptions, this component was meant to disable reduced motion, but you are right. It could be confusing that we can pass more types of actions here. I have implemented it in this [commit](https://github.com/software-mansion/react-native-reanimated/pull/6164/commits/3bf2eea1bf9f11b54f590ce1abd61ab6864f71d3).